vignettes/models.Rmd

```diff
@@ -0,0 +1,322 @@
+---
+title: "Model Families in sgdnet"
+author: "Johan Larsson"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+bibliography: "sgdnet.bib"
+vignette: >
+  %\VignetteIndexEntry{Model Families in sgdnet}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+
+library(lattice)
+```
+
+## Overview
+
+**sgdnet** fits generalized linear models of the type
+
+$$
+\min_{\beta_0, \beta}
+\left\{
+-\frac1n \mathcal{L}\left(\beta_0,\beta; \mathbf{y}, \mathbf{X}\right)
++ \lambda \left[(1 - \alpha)||\beta||_2^2 + \alpha||\beta||_1 \right]
+\right\},
+$$
+where $\mathcal{L}(\beta_0,\beta; \mathbf{y}, \mathbf{X})$ is the 
+log-likelihood of the model, $\lambda$ is the regularization strength, and
+$\alpha$, is the elastic net mixing parameter [@zou2005], such that
+$\alpha = 1$ results in the lasso [@tibshirani1996] and $\alpha = 0$ the ridge
+penalty.
```

Comment:
Cite Hoerl and Kennard (1970, Technometrics) for ridge (at least in the statistics context - the idea dates back to Tikhonov in the 1940s)